src/drop.ts

```diff
@@ -1,6 +1,16 @@
+import { type ArraySlice } from "type-fest";
 import { SKIP_ITEM, lazyIdentityEvaluator } from "./internal/utilityEvaluators";
 import type { LazyEvaluator } from "./pipe";
 import { purry } from "./purry";
+import { type IterableContainer } from "./internal/types";
+
+type Drop<T extends IterableContainer, N extends number> = number extends N
+  ? Array<T[number]>
+  : `${N}` extends `-${number}`
+    ? T
+    : number extends T["length"]
+      ? Array<T[number]>
+      : ArraySlice<T, N, T["length"]>;
```

Comment:
This solution doesn't solve the most useful case for a better type: when an array has a variable length but still has some prefix we can maintain; for example: `[string, string, string, ...Array<string>]`. In this case, I would expect `drop` with an `N` of 1 to return the type `[string, string, ...Array<string>]`, which your solution doesn't do.

Some limitations with type-fest's `ArraySlice` (and `ArraySplice`) don't make them an easy tool to solve this. I know we suggested using them in previous review comments, but I now think that was a mistake.

Solving `drop's return type has been requested in the past, and I pushed against it because it was difficult.

Fortunately, we recently added some utilities that make the type for drop easier. The solution is still complex, so I prefer just providing the solution, but if you have questions to better understand it, feel free.

```suggestion
    : [
        ...(TupleParts<T>["prefix"] extends readonly [
          ...NTuple<unknown, N>,
          ...infer V,
        ]
          ? V
          : []),
        ...IfNever<
          TupleParts<T>["item"],
          [],
          [...Array<TupleParts<T>["item"]>, ...TupleParts<T>["suffix"]]
        >,
      ];
```

Reply:
Done